### DISCUSSION TITLE
Convert `annotateGEM` into a more generic function

### DISCUSSION THREAD

**Discussion by haowang-bioinfo** [SysBioChalmers/Human-GEM#380] [Ideas]
The `annotateGEM` function has been embedded in `exportHumanGEM` function and used to integrate annotation information into Human-GEM before exporting to `xml` format. Would be nice to make it more generic so that it could be used by other models, e.g. Mouse-GEM.

Following adjustments are proposed for making `annotateGEM` a more generic function:
- introduce an additional argument `annotationPath` to specify the location of annotation files
- set `annotationPath` as the 2nd argument, only two argument are required when used for other models
- if this argument is missing, fetch annotation files from default path of Human-GEM, as is
- there is no changes to any other functions that call this function

**Comment by JonathanRob** [comment]
Good suggestion, @haowang-bioinfo. I guess the `annotationPath` could directly reference the file on GitHub (e.g., `annotationPath = https://raw.githubusercontent.com/SysBioChalmers/Mouse-GEM/main/model/`), or be a relative/local path, right?

**Comment by haowang-bioinfo** [comment]
> I guess the annotationPath could directly reference the file on GitHub (e.g., annotationPath = https://raw.githubusercontent.com/SysBioChalmers/Mouse-GEM/main/model/), or be a relative/local path, right?

thanks @JonathanRob for the feedback. I only considered local path when proposing `annotationPath` argument. Really uncertain about how difficult in enabling it also as a cloud path.

**Comment by JonathanRob** [comment]
I think it's possible using `urlread`, but now that I think about it, probably not worth the added work of enabling a web path. I agree that a local path is the most useful option for now.

**Comment by haowang-bioinfo** [comment]
> I think it's possible using `urlread`, but now that I think about it, probably not worth the added work of enabling a web path. I agree that a local path is the most useful option for now.

thanks, will work on the implementation of adding `annotationPath` argument for local path and keep the web path option open for future refinement.
